src/find_bottleneck.rs

```diff
@@ -8,37 +8,56 @@ pub(crate) struct FindBottleneck {
 
 impl FindBottleneck {
     /// Creates a new `FindBottleneck`, reads and parses mrt files, locates prefix and asn bottleneck
-    pub(crate) fn locate(dir: &PathBuf) -> Result<Self> {
-        let mut mrt_hm = HashMap::new();
+    pub(crate) fn locate(dir: &PathBuf) -> Result<()> {
+        if !dir.is_dir() {
+            Error::NotDirError{path: dir.to_path_buf()};
+        }
+
+        let mut file_decoders = Vec::new();
+
         // Walk the directory and read its contents
-        if dir.is_dir() {
-            for entry in fs::read_dir(dir).map_err(|io_error| Error::IoError {
+        for entry in fs::read_dir(dir).map_err(|io_error| Error::IoError {
+            io_error,
+            path: "path".into(),
+        })? {
+            let entry = entry.map_err(|io_error| Error::IoError {
                 io_error,
                 path: "path".into(),
-            })? {
-                let entry = entry.map_err(|io_error| Error::IoError {
+            })?;
+            let path = entry.path();
+            println!("Acquiring a reader for file`{}`", &path.display());
+            let buffer =
+                BufReader::new(File::open(&path).map_err(|io_error| Error::IoError {
                     io_error,
-                    path: "path".into(),
-                })?;
-                let path = entry.path();
-                println!("Reading in and parsing `{}`", &path.display());
-                let buffer =
-                    BufReader::new(File::open(&path).map_err(|io_error| Error::IoError {
-                        io_error,
-                        path: path.into(),
-                    })?);
-
-                let mut decoder = GzDecoder::new(buffer);
-                Self::parse_mrt(&mut decoder, &mut mrt_hm)?;
-            }
+                    path: path.into(),
+                })?);
+            
+            let decoder = GzDecoder::new(buffer);
+            file_decoders.push(decoder);
         }
 
-        let mut bottleneck = FindBottleneck {
-            prefix_asn: HashMap::new(),
-        };
-        bottleneck.find_as_bottleneck(&mut mrt_hm)?;
+        let step: u8 = 16;
+        assert!(step.is_power_of_two());
```

Comment:
It is a little odd to use an assert here I think. Enforce this in the code by using `1 << 4`.